### ISSUE TITLE
Intersect

### ISSUE THREAD

**Issue by isc30** [isc30/linq-collections#19]
Produces the set intersection of two sequences (deferred execution)

Proposed signature:

```ts
export interface IQueryable<TOut>
{
    intersect(other: IQueryable<TOut>): IEnumerable<TOut>;
}
```

**Comment by nikolalukovic** [comment]
I suppose we will also need to have something like an `EqualityComparer` to specify in which way to compare values?

**Comment by isc30** [comment]
The only place where comparers are being used currently is for `orderBy`, maybe it's a good idea to introduce a `StrictEqualityComparer` that simply does `===` and start using it here as default one?

**Comment by nikolalukovic** [comment]
It all depends how precise you want to go. I'd argue that for strings, as an example, being careful about locale is a must. And `===` doesn't help in that case.

**Comment by isc30** [comment]
yep, but in the end, this is user's responsibility. I think the best we can do is use `===` as default (every other method in the library uses this comparer)
